contracts/orbital-core/src/account_types.rs

```diff
@@ -0,0 +1,31 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{ensure, DepsMut, StdError, StdResult, Uint64};
+
+use crate::state::OrbitalDomainConfig;
+
+#[cw_serde]
+pub enum AccountConfigType {
+    Polytone { note: String, timeout: Uint64 },
+}
+
+impl AccountConfigType {
+    pub fn try_into_domain_config(&self, deps: &DepsMut) -> StdResult<OrbitalDomainConfig> {
```

Comment:
I wanted to have a way to validate the fields that can be validated before saving it in storage. e.g. `note`/`mailbox` addresses in case of polytone and hyperlane respectively, timeouts being non-zero, etc. Perhaps naming it with  _unchecked_ prefix would make it more clean, like `UncheckedOrbitalDomainConfig`, and the  `to_checked()` method then producing `OrbitalDomainConfig` would be cleaner?

Reply:
Yea, the renaming should help and explain better what you trying to do